lib/ansible/cli/galaxy.py

```diff
@@ -916,8 +947,71 @@ def execute_init(self):
         display.display("- %s %s was created successfully" % (galaxy_type.title(), obj_name))
 
     def execute_info(self):
+        if context.CLIARGS['type'] == 'role':
+            self.execute_info_role()
+        elif context.CLIARGS['type'] == 'collection':
+            self.execute_info_collection()
+
+    def execute_info_collection(self):
+        """Find and display the metadata for a given collection."""
+
+        collection_name = context.CLIARGS['args'][0]
+        collections_search_paths = set(context.CLIARGS['collections_path'])
+        default_collections_path = C.config.get_configuration_definition('COLLECTIONS_PATHS').get('default')
+
+        warnings = []
+        path_found = False
+        collection_found = False
+
+        for path in collections_search_paths:
+            collection_path = GalaxyCLI._resolve_path(path)
+            if not os.path.exists(path):
+                if path in default_collections_path:
+                    # don't warn for missing default paths
+                    continue
+                warnings.append("- the configured path {0} does not exist.".format(collection_path))
+                continue
+
+            if not os.path.isdir(collection_path):
+                warnings.append("- the configured path {0}, exists, but it is not a directory.".format(collection_path))
+                continue
+
+            path_found = True
+
+            validate_collection_name(collection_name)
+            namespace, collection = collection_name.split('.')
```

Comment:
Great to know! Will fix that up.

Reply:
Since collection_name is coming in via the command line (which does not support versions), I think this is ok here. validate_collection_name will look for a version if one exists, but it will return the unmodified value that was passed in if it is valid (raises an exception otherwise).

Reply:
Doesn't the command line support versions? Via the `:version` syntax. For example `ansible-galaxy collection install amazon.aws:1.0.0` installs 1.0.0 instead of the latest 1.0.1-dev1.

Reply:
Ah, I was considering it in the context of 'collection list', but yeah, if 'install' supports a version, it probably makes sense for 'collection info' to support it too. Thanks for pointing that out!